ruby/lib/clone_handler.rb

```diff
@@ -0,0 +1,122 @@
+require 'json'
+require 'net/http'
+
+class CloneHandler
+  PER_PAGE = 30.freeze
+  BASE_URL = 'https://api.github.com'
+
+  def self.get_username
+    Kernel.puts 'Please enter your Github username'
+    STDOUT.flush
+    STDIN.gets.chomp
+  end
+
+  def initialize(username)
+    @username = username
+    @errors = []
+  end
+
+  def clone
+    validate_clonable
```

Comment:
Validate that we have everything before cloning. If something is missing or an API request failed, it will be added to an errors list and printed.